xcm-simulator/src/lib.rs

```diff
@@ -0,0 +1,689 @@
+			use asset_reserve::{MsgQueue, Runtime, System};
+
+			const INITIAL_BALANCE: u128 = <Runtime as pallet_assets::Config>::AssetDeposit::get() * 2;
+
+			let mut t = frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+
+			pallet_balances::GenesisConfig::<Runtime> { balances: vec![
+					(ALICE, INITIAL_BALANCE),
+					(asset_reserve::sovereign_account(TRAPPIST_PARA_ID), INITIAL_BALANCE)
+				]}
+				.assimilate_storage(&mut t)
+				.unwrap();
+
+			let mut ext = sp_io::TestExternalities::new(t);
+			ext.execute_with(|| {
+				sp_tracing::try_init_simple();
+				System::set_block_number(1);
+				MsgQueue::set_para_id(ASSET_RESERVE_PARA_ID.into());
+			});
+			ext
+		},
+	}
+}
+
+const TRAPPIST_PARA_ID: u32 = 2000;
+decl_test_parachain! {
+	// The trappist parachain
+	pub struct Trappist {
+		Runtime = trappist::Runtime,
+		XcmpMessageHandler = trappist::MsgQueue,
+		DmpMessageHandler = trappist::MsgQueue,
+		new_ext = {
+			// Initialise parachain-specific genesis state
+			use trappist::{MsgQueue, Runtime, System};
+
+			let asset_deposit: u128 = <trappist::Runtime as pallet_assets::Config>::AssetDeposit::get();
+			let initial_balance: u128 = asset_deposit * 2;
+
+			let mut t = frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+
+			pallet_balances::GenesisConfig::<Runtime> { balances: vec![(ALICE, initial_balance)] }
+				.assimilate_storage(&mut t)
+				.unwrap();
+
+			pallet_sudo::GenesisConfig::<Runtime> { key: Some(ALICE) }
+				.assimilate_storage(&mut t)
+				.unwrap();
+
+			let mut ext = sp_io::TestExternalities::new(t);
+			ext.execute_with(|| {
+				sp_tracing::try_init_simple();
+				System::set_block_number(1);
+				MsgQueue::set_para_id(TRAPPIST_PARA_ID.into());
+			});
+			ext
+		},
+	}
+}
+
+const BASE_PARA_ID: u32 = 3000;
+decl_test_parachain! {
+	// A parachain using the trappist 'base' runtime
+	pub struct Base {
+		Runtime = base::Runtime,
+		XcmpMessageHandler = base::MsgQueue,
+		DmpMessageHandler = base::MsgQueue,
+		new_ext = {
+			// Initialise parachain-specific genesis state
+			use trappist::{MsgQueue, Runtime, System};
+
+			let asset_deposit: u128 = <trappist::Runtime as pallet_assets::Config>::AssetDeposit::get();
+			let initial_balance: u128 = asset_deposit * 2;
+
+			let mut t = frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+
+			pallet_balances::GenesisConfig::<Runtime> { balances: vec![(ALICE, initial_balance)] }
+				.assimilate_storage(&mut t)
+				.unwrap();
+
+			pallet_sudo::GenesisConfig::<Runtime> { key: Some(ALICE) }
+				.assimilate_storage(&mut t)
+				.unwrap();
+
+			let mut ext = sp_io::TestExternalities::new(t);
+			ext.execute_with(|| {
+				sp_tracing::try_init_simple();
+				System::set_block_number(1);
+				MsgQueue::set_para_id(BASE_PARA_ID.into());
+			});
+			ext
+		},
+	}
+}
+
+const A_PARA_ID: u32 = 1;
+decl_test_parachain! {
+	// Some generic parachain
+	pub struct ParaA {
```

Comment:
what's the role of `ParaA`?

Reply:
I'd give the variable a more insightful name, and also a more detailed description in the comments.

Reply:
Renamed to `Template` to better describe intention, along with improved comments